bin/flash.sdwire_digital-loggers_mount

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Google LLC
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+# Designed for using SDwire uSD mux and ykush USB power control:
+# https://wiki.tizen.org/SDWire
+# https://www.yepkit.com/products/ykush
+
+# It requires these tools:
+# sd-mux-ctrl at https://git.tizen.org/cgit/tools/testlab/sd-mux/
+# curl to control the power switch
+
+# Power off
+curl --data ${power_port}=OFF -o /dev/null --silent \
```

Comment:
Does this power off the target, which in turn means that the console UART USB device goes away? I don't think that will work right, since if the console device goes away, picocom will error out, which will cause the test to fail. Does this all work, and if so, are there failures reported in the log file where picocom dies and perhaps is restarted via a target reset/... ?

Reply:
BTW, the expected model is that power is first turned on and left on, then the test script runs completely (including any flashing or resets required), then power is turned off. Typically I'd expect the implementation of flashing/reset/... to force the board into reset or other disabled/recovery/... mode rather than ever power it off in the middle of the test. That should prevent the UART USB device from going away, since there'd never be a power loss.

Reply:
OK but what if I don't have reset control? I need to use a power cycle in that case.

Reply:
If you have no control over reset so you have to use power instead *and* the UART USB device goes away when you power cycle, then that won't work well!

Ideally, you'd wire up a relay/... to the reset control so that reset is possible.

Otherwise, about the only thing I can think of is to write a custom console hook implementation that ignores missing devices; perhaps something like:

```
while true; do
    echo "CONSOLE: WAITING FOR CONSOLE DEVICE"
    while [ ! -e "${console_dev}" ]; then
        sleep 1
    done
    echo "CONSOLE: STARTING PICOCOM"
    picocom -b "${console_baud}" "${console_dev}"
    echo "CONSOLE: PICOCOM EXITED!"
done
```

That way, you can power cycle the device yet the test core won't be aware of it, since the console implementation never exits. However, you still run the risk of losing many seconds of initial console output while the USB UART is enumerated and before the script above sees the device and starts picocom. Perhaps the USB UART chip will buffer some of that data though, but I don't image the buffer is *that* large even if there is one.